pkg/services/ngalert/api/tooling/definitions/alertmanager.go

```diff
@@ -479,6 +482,50 @@ type GettableSilences = amv2.GettableSilences
 // swagger:model gettableSilence
 type GettableSilence = amv2.GettableSilence
 
+// swagger:model gettableGrafanaSilence
+type GettableGrafanaSilence struct {
+	*GettableSilence `json:",inline"`
+	Metadata         *SilenceMetadata `json:"metadata,omitempty"`
+}
+
+type SilenceMetadata struct {
+	RuleUID     string                         `json:"rule_uid,omitempty"`
+	RuleTitle   string                         `json:"rule_title,omitempty"`
+	FolderUID   string                         `json:"folder_uid,omitempty"`
+	Permissions map[SilencePermission]struct{} `json:"permissions,omitempty"`
```

Comment:
Should discuss what format we want to return here. Another option could be `map[SilencePermission]bool` or something else if it's more inline with how we return permissions elsewhere.

Reply:
Yes, I think it'd be more consistent if this returned a map of actions like you said 🙏  This is more consistent with the folder endpoint.

If we wanted to be even more consistent with elsewhere, then the permissions would only be returned if a query param `accesscontrol` was sent - this is how the folder endpoint works, but it doesn't matter too much to me as the UI will be sending the param(s) to get all the metadata on all requests it makes to the silences endpoint

Reply:
Modified the query params and response as you suggested:

Replaces `withMetadata` query param with separate:
- `ruleMetadata`: to request rule metadata.
- `accesscontrol`: to request access control metadata.

Also modifies access control metadata response to be:
`"accessControl": map[string]bool`

Updated the example response in the PR description.

Reply:
Sorry, my fault 🤦 I don't think I was clear enough about the structure - could the metadata/access control object be separate from each other?

So, asking for metadata gives me
```ts
{
  //...normal silence stuff
  id: 'foo',

  metadata: {
    rule_name: 'bar',
  },
}
```

Asking for ACL gives me
```ts
{
  //...normal silence stuff
  id: 'foo',

  accessControl: {
    write: true
    // etc
  }
}
```


So they work independently from each other?